probability/windowed_median.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+ * @file
+ * @brief An implementation of a median calculation of a sliding window along a data stream
+ *
+ * @details
+ * Given a stream of integers, the algorithm calculates the median of a fix size window at the back of the stream. The leading time complexity of this algorithm is
+ * O(log(N), and it is inspired by the known algorithm to [find median from (infinite) data stream](https://www.tutorialcup.com/interview/algorithm/find-median-from-data-stream.htm), with the proper modifications to account
+ * for the finite window size for which the median is requested
+ *
+ * ### Algorithm
+ * The sliding window is managed by a list, which guarantees O(1) for both pushing and popping. Each new value is pushed to the window back, while a value
+ * from the front of the window is popped. In addition, the algorithm manages a multi-value binary search tree (BST), implemented by std::multiset. For each new
+ * value that is inserted to the window, it is also inserted to the BST. When a value is popped from the window, it is also erased from the BST. Both insertion and
+ * erasion to/from the BST are O(logN) in time, with N the size of the window. Finally, the algorithm keeps a pointer to the root of the BST, and updates its position
+ * whenever values are inserted or erased to/from BST. The root of the tree is the median! Hence, median retrieval is always O(1)
+ *
+ * Time complexity: O(logN). Space complexity: O(N). N - size of window
+ * @author [Yaniv Hollander](https://github.com/YanivHollander)
+ */
+#include <cassert>      // for assert
+#include <cstdint>      // for std::uint64_t
```

Comment:
I'm not sure if this one is required. AFAIK, you can just use the `uint64_t` variable without the need of any headers or with `std::` at the beginning. 🙂

Reply:
Well, I guess you are right. But actually, is using `uint64_t` concise in this case? Please review my revision. I changed to using the `size_type` of std::list as the window size type.